// @bit/grommet.grommet.utils.styles/dist/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sizeStyle = exports.disabledStyle = exports.genericStyles = exports.placeholderStyle = exports.overflowStyle = exports.inputStyle = exports.focusStyle = exports.edgeStyle = exports.controlBorderStyle = exports.baseStyle = void 0;

var _styledComponents = require("styled-components");

var _background = require("./background");

var _colors = require("./colors");

var _mixins = require("./mixins");

function _templateObject21() {
  var data = _taggedTemplateLiteral(["\n  ", ": ", ";\n"]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = _taggedTemplateLiteral(["\n  opacity: ", ";\n  cursor: default;\n"]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = _taggedTemplateLiteral(["\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = _taggedTemplateLiteral(["\n  &::-webkit-input-placeholder {\n    ", ";\n  }\n\n  &::-moz-placeholder {\n    ", ";\n  }\n\n  &:-ms-input-placeholder {\n    ", ";\n  }\n"]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = _taggedTemplateLiteral(["\n  color: ", ";\n"]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = _taggedTemplateLiteral(["\n    ", " ", ";\n  "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = _taggedTemplateLiteral(["\n      overflow: ", ";\n    "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = _taggedTemplateLiteral(["\n      font-weight: ", ";\n    "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteral(["\n  box-sizing: border-box;\n  font-size: inherit;\n  font-family: inherit;\n  border: none;\n  -webkit-appearance: none;\n  padding: ", "px;\n  outline: none;\n  background: transparent;\n  color: inherit;\n  ", " margin: 0;\n\n  ", " ", "\n\n  ::-webkit-search-decoration {\n    -webkit-appearance: none;\n  }\n"]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteral(["\n  > circle,\n  > ellipse,\n  > line,\n  > path,\n  > polygon,\n  > polyline,\n  > rect {\n    outline: ", "\n      solid 2px;\n  }\n  outline-color: ", ";\n  border-color: ", ";\n  box-shadow: 0 0 2px 2px\n    ", ";\n\n  ::-moz-focus-inner {\n    border: 0;\n  }\n"]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteral(["\n      ", "-inline-end: ", ";\n      ", ";\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n      ", "-inline-start: ", ";\n      ", ";\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n      ", "-right: ", ";\n      ", ";\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n      ", "-left: ", ";\n      ", ";\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n      ", "-bottom: ", ";\n      ", ";\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n      ", "-top: ", ";\n      ", ";\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n      ", "-top: ", ";\n      ", "-bottom: ", ";\n      ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n      ", "-left: ", ";\n      ", "-right: ", ";\n      ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n      ", ": ", ";\n      ", ";\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  border: ", " solid\n    ", ";\n  border-radius: ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n  font-size: ", ";\n  line-height: ", ";\n  font-weight: ", ";\n  ", "\n  box-sizing: border-box;\n  -webkit-text-size-adjust: 100%;\n  -ms-text-size-adjust: 100%;\n  -moz-osx-font-smoothing: grayscale;\n  -webkit-font-smoothing: antialiased;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var baseStyle = (0, _styledComponents.css)(_templateObject(), function (props) {
  return props.theme.global.font.family;
}, function (props) {
  return props.theme.global.font.size;
}, function (props) {
  return props.theme.global.font.height;
}, function (props) {
  return props.theme.global.font.weight;
}, function (props) {
  return !props.plain && (0, _background.backgroundStyle)(props.theme.baseBackground, props.theme);
});
exports.baseStyle = baseStyle;
var controlBorderStyle = (0, _styledComponents.css)(_templateObject2(), function (props) {
  return props.theme.global.control.border.width;
}, function (props) {
  return (0, _colors.normalizeColor)(props.theme.global.control.border.color || 'border', props.theme);
}, function (props) {
  return props.theme.global.control.border.radius;
});
exports.controlBorderStyle = controlBorderStyle;

var edgeStyle = function edgeStyle(kind, data, responsive, responsiveBreakpoint, theme) {
  var breakpoint = responsiveBreakpoint && theme.global.breakpoints[responsiveBreakpoint];

  if (typeof data === 'string') {
    return (0, _styledComponents.css)(_templateObject3(), kind, theme.global.edgeSize[data] || data, responsive && breakpoint ? (0, _mixins.breakpointStyle)(breakpoint, "\n        ".concat(kind, ": ").concat(breakpoint.edgeSize[data] || data, ";\n      ")) : '');
  }

  var result = [];

  if (data.horizontal) {
    result.push((0, _styledComponents.css)(_templateObject4(), kind, theme.global.edgeSize[data.horizontal] || data.horizontal, kind, theme.global.edgeSize[data.horizontal] || data.horizontal, responsive && breakpoint ? (0, _mixins.breakpointStyle)(breakpoint, "\n        ".concat(kind, "-left: ").concat(breakpoint.edgeSize[data.horizontal] || data.horizontal, ";\n        ").concat(kind, "-right: ").concat(breakpoint.edgeSize[data.horizontal] || data.horizontal, ";\n      ")) : ''));
  }

  if (data.vertical) {
    result.push((0, _styledComponents.css)(_templateObject5(), kind, theme.global.edgeSize[data.vertical] || data.vertical, kind, theme.global.edgeSize[data.vertical] || data.vertical, responsive && breakpoint ? (0, _mixins.breakpointStyle)(breakpoint, "\n        ".concat(kind, "-top: ").concat(breakpoint.edgeSize[data.vertical] || data.vertical, ";\n        ").concat(kind, "-bottom: ").concat(breakpoint.edgeSize[data.vertical] || data.vertical, ";\n      ")) : ''));
  }

  if (data.top) {
    result.push((0, _styledComponents.css)(_templateObject6(), kind, theme.global.edgeSize[data.top] || data.top, responsive && breakpoint ? (0, _mixins.breakpointStyle)(breakpoint, "\n        ".concat(kind, "-top: ").concat(breakpoint.edgeSize[data.top] || data.top, ";\n      ")) : ''));
  }

  if (data.bottom) {
    result.push((0, _styledComponents.css)(_templateObject7(), kind, theme.global.edgeSize[data.bottom] || data.bottom, responsive && breakpoint ? (0, _mixins.breakpointStyle)(breakpoint, "\n        ".concat(kind, "-bottom: ").concat(breakpoint.edgeSize[data.bottom] || data.bottom, ";\n      ")) : ''));
  }

  if (data.left) {
    result.push((0, _styledComponents.css)(_templateObject8(), kind, theme.global.edgeSize[data.left] || data.left, responsive && breakpoint ? (0, _mixins.breakpointStyle)(breakpoint, "\n        ".concat(kind, "-left: ").concat(breakpoint.edgeSize[data.left] || data.left, ";\n      ")) : ''));
  }

  if (data.right) {
    result.push((0, _styledComponents.css)(_templateObject9(), kind, theme.global.edgeSize[data.right] || data.right, responsive && breakpoint ? (0, _mixins.breakpointStyle)(breakpoint, "\n        ".concat(kind, "-right: ").concat(breakpoint.edgeSize[data.right] || data.right, ";\n      ")) : ''));
  }

  if (data.start) {
    result.push((0, _styledComponents.css)(_templateObject10(), kind, theme.global.edgeSize[data.start] || data.start, responsive && breakpoint ? (0, _mixins.breakpointStyle)(breakpoint, "\n        ".concat(kind, "-inline-start: ").concat(breakpoint.edgeSize[data.start] || data.start, ";\n      ")) : ''));
  }

  if (data.end) {
    result.push((0, _styledComponents.css)(_templateObject11(), kind, theme.global.edgeSize[data.end] || data.end, responsive && breakpoint ? (0, _mixins.breakpointStyle)(breakpoint, "\n        ".concat(kind, "-inline-end: ").concat(breakpoint.edgeSize[data.end] || data.end, ";\n      ")) : ''));
  }

  return result;
}; // focus also supports clickable elements inside svg


exports.edgeStyle = edgeStyle;
var focusStyle = (0, _styledComponents.css)(_templateObject12(), function (props) {
  return (0, _colors.normalizeColor)(props.theme.global.focus.border.color, props.theme);
}, function (props) {
  return (0, _colors.normalizeColor)(props.theme.global.focus.border.color, props.theme);
}, function (props) {
  return (0, _colors.normalizeColor)(props.theme.global.focus.border.color, props.theme);
}, function (props) {
  return (0, _colors.normalizeColor)(props.theme.global.focus.border.color, props.theme);
});
exports.focusStyle = focusStyle;
var inputStyle = (0, _styledComponents.css)(_templateObject13(), function (props) {
  return (0, _mixins.parseMetricToNum)(props.theme.global.input.padding) - (0, _mixins.parseMetricToNum)(props.theme.global.control.border.width);
}, function (props) {
  return props.theme.global.input.weight && (0, _styledComponents.css)(_templateObject14(), props.theme.global.input.weight);
}, function (props) {
  return props.focus && (!props.plain || props.focusIndicator) && focusStyle;
}, controlBorderStyle);
exports.inputStyle = inputStyle;

var overflowStyle = function overflowStyle(overflowProp) {
  if (typeof overflowProp === 'string') {
    return (0, _styledComponents.css)(_templateObject15(), overflowProp);
  }

  return (0, _styledComponents.css)(_templateObject16(), overflowProp.horizontal && "overflow-x: ".concat(overflowProp.horizontal, ";"), overflowProp.vertical && "overflow-y: ".concat(overflowProp.vertical, ";"));
};

exports.overflowStyle = overflowStyle;
var placeholderColor = (0, _styledComponents.css)(_templateObject17(), function (props) {
  return props.theme.global.colors.placeholder;
});
var placeholderStyle = (0, _styledComponents.css)(_templateObject18(), placeholderColor, placeholderColor, placeholderColor);
exports.placeholderStyle = placeholderStyle;
var ALIGN_SELF_MAP = {
  center: 'center',
  end: 'flex-end',
  start: 'flex-start',
  stretch: 'stretch'
};
var genericStyles = (0, _styledComponents.css)(_templateObject19(), function (props) {
  return props.alignSelf && "align-self: ".concat(ALIGN_SELF_MAP[props.alignSelf], ";");
}, function (props) {
  return props.gridArea && "grid-area: ".concat(props.gridArea, ";");
}, function (props) {
  return props.margin && props.theme.global && edgeStyle('margin', props.margin, props.responsive, props.theme.global.edgeSize.responsiveBreakpoint, props.theme);
});
exports.genericStyles = genericStyles;

var disabledStyle = function disabledStyle(componentStyle) {
  return (0, _styledComponents.css)(_templateObject20(), function (props) {
    return componentStyle || props.theme.global.control.disabled.opacity;
  });
};

exports.disabledStyle = disabledStyle;

var sizeStyle = function sizeStyle(name, value, theme) {
  return (0, _styledComponents.css)(_templateObject21(), name, theme.global.size[value] || value);
};

exports.sizeStyle = sizeStyle;
