// @bit/tabinda.react-button-animation.animated.button/dist/Button.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ButtonHover = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

require("./ButtonStyles.css");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ButtonHover = function ButtonHover(props) {
  var className = props.className,
      onClick = props.onClick,
      label = props.label,
      isDisabled = props.isDisabled;
  return _react.default.createElement("button", {
    onClick: onClick
  }, _react.default.createElement("span", {
    className: className,
    disabled: isDisabled
  }, _react.default.createElement("span", null, label), _react.default.createElement("svg", {
    width: "13px",
    height: "10px",
    viewBox: "0 0 13 10"
  }, _react.default.createElement("path", {
    d: "M1,5 L11,5"
  }), _react.default.createElement("polyline", {
    points: "8 1 12 5 8 9"
  }))));
};

exports.ButtonHover = ButtonHover;
ButtonHover.propTypes = {
  className: _propTypes.default.string,
  onClick: _propTypes.default.func,
  label: _propTypes.default.string,
  isDisabled: _propTypes.default.bool
};
ButtonHover.defaultProps = {
  className: "cta",
  onClick: null,
  label: "Click Me",
  isDisabled: false
};
var _default = ButtonHover;
exports.default = _default;
